src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -1794,16 +1802,15 @@ private void EnterParameters(ImmutableArray<ParameterSymbol> parameters)
 
         protected virtual void EnterParameter(ParameterSymbol parameter)
         {
+            int slot = GetOrCreateSlot(parameter);
             if (parameter.RefKind == RefKind.Out && !this.currentMethodOrLambda.IsAsync) // out parameters not allowed in async
             {
-                int slot = GetOrCreateSlot(parameter);
                 if (slot > 0) SetSlotState(slot, initiallyAssignedVariables?.Contains(parameter) == true);
             }
             else
             {
                 // this code has no effect except in region analysis APIs such as DataFlowsOut where we unassign things
-                int slot = GetOrCreateSlot(parameter);
-                if (slot > 0) SetSlotState(slot, true);
+                if (slot > 0) SetSlotState(slot, !_trackClasses || parameter != MethodThisParameter);
                 NoteWrite(parameter, value: null, read: true);
```

Comment:
>NoteWrite(parameter, value: null, read: true); [](start = 16, length = 46)

Should this line be adjusted in any way?

Reply:
Reverted change.